lib/MongoDB/ServerSession.pm

```diff
@@ -50,26 +61,64 @@ sub _build_session_id {
     return { id => $uuid };
 }
 
+=method last_use
+
+    $sever_session->last_use;
+
+Returns the unix time that this server session was last used. Used for checking
+expiry of a server session. If undefined, then the session has (probably) not
+been used on the server.
+
+=cut
+
 has last_use => (
     is => 'rwp',
     init_arg => undef,
-    isa => Maybe[InstanceOf['DateTime']],
+    isa => Maybe[Int],
 );
 
+=method update_last_use
+
+    $server_session->update_last_use;
+
+Updates the value of L</last_use> to the current unix time.
+
+=cut
+
 sub update_last_use {
     my ( $self ) = @_;
-    $self->_set_last_use( DateTime->now );
+    $self->_set_last_use( time() );
 }
 
 sub _is_expiring {
     my ( $self, $session_timeout ) = @_;
 
-    my $timeout = DateTime->now;
-    $timeout->subtract( minutes => $session_timeout - 1 );
+    # if session_timeout is undef, then sessions arent actually supported (this
+    # value should be from logical_session_timeout_minutes).
+    return 1 unless defined $session_timeout;
+
+    my $timeout = time() - ( ( $session_timeout - 1 ) * 60 );
 
     # Undefined last_use means its never actually been used on the server
     return 1 if defined $self->last_use && $self->last_use < $timeout;
     return;
 }
 
 1;
+
+__END__
+
+=pod
+
+=head1 SYNOPSIS
+
+    use MongoDB::ServerSession;
+
+    my $server_session = MongoDB::ServerSession->new;
+
+=head1 DESCRIPTION
+
+This class encapsulates the session id and last use of the session. For use
+with L<MongoDB::ClientSession> for session based operations.
```

Comment:
Seeing this makes me think we should make this a private MongoDB::_ServerSession class instead of a public one.  Is there any user-facing need for it?

Reply:
Hmm fair point, there's really no reason a user should need to create one of these. Will swap it over to private naming